eth/catalyst/simulated_beacon.go

```diff
@@ -261,6 +271,48 @@ func (c *SimulatedBeacon) setCurrentState(headHash, finalizedHash common.Hash) {
 	}
 }
 
+// Commit seals a block on demand.
+func (c *SimulatedBeacon) Commit() common.Hash {
+	withdrawals := c.withdrawals.gatherPending(10)
+	if err := c.sealBlock(withdrawals, uint64(time.Now().Unix())); err != nil {
+		log.Warn("Error performing sealing work", "err", err)
+	}
+	return c.eth.BlockChain().CurrentBlock().Hash()
+}
+
+// Rollback un-sends previously added transactions.
+func (c *SimulatedBeacon) Rollback() {
+	// TODO (MariusVanDerWijden): implement a way in the transaction pool
+	// to flush _all_ transactions, even local transactions.
+	// Since the transactions send via the simulated backend are all
+	// send via the RPC, they end up all being local to the node
```

Comment:
Perhaps an easy trick to make it work is configuring txpool with `NoLocal = true`.

Reply:
Good idea, I reverted the change and set NoLocal to true when constructing a simulated backend